db/compaction_job.cc

```diff
@@ -171,12 +171,18 @@ struct CompactionJob::SubcompactionState {
     const std::vector<FileMetaData*>& grandparents = compaction->grandparents();
```

Comment:
Not necessary to address for this PR - I feel like this function should operate at the granularity of atomic compaction unit rather than individual files. Particularly now that we cut to avoid overlapping a unit (currently a file), it is only beneficial if that unit also happens to be an atomic compaction unit, meaning it would not get pulled into its neighbors' compactions.